### PR TITLE
Handle int64 columns with missing data in SQL Lab

### LINKED ISSUE #8225: Pandas casting int64 to float64, misrepresenting value
I have the following data being returned by Presto (single column, 6 rows):

```
[(None,), (1239162456494753670,), (None,), (None,), (None,), (None,)
```

Due to the missing data (`None`), Pandas infers the type as `float64`, converting the value to a wrong id:

```python
>>> column_names = ['organization_lyft_id']
>>> data = [(None,), (1239162456494753670,), (None,), (None,), (None,), (None,)]
>>> df = pd.DataFrame(list(data), columns=column_names).infer_objects()  # SupersetDataFrame
>>> print(df)
   organization_lyft_id
0                   NaN
1          1.239162e+18
2                   NaN
3                   NaN
4                   NaN
5                   NaN
>>> print(df.dtypes)
organization_lyft_id    float64
dtype: object
```

The number then shows up as `1239162456494753800` in SQL Lab.

Here's the Pandas documentation on this:

> ... pandas primarily uses NaN to represent missing data. Because NaN is a float, this forces an array of integers with any missing values to become floating point. In some cases, this may not matter much. But if your integer column is, say, an identifier, casting to float can be problematic. **Some integers cannot even be represented as floating point numbers.** (emphasis mine)

Note that if the missing data is filtered the value is inferred as an int64, and it shows up correctly in SQL Lab:

```python
>>> column_names = ['organization_lyft_id']
>>> data = [(1239162456494753670,)]
>>> df = pd.DataFrame(list(data), columns=column_names).infer_objects()  # SupersetDataFrame
>>> print(df)
   organization_lyft_id
0   1239162456494753670
>>> print(df.dtypes)
organization_lyft_id    int64
dtype: object

```

The solution is to pass a `dtype` argument when creating the Pandas data frame, built from the cursor description. I'm working on a fix for this.

### PR DESCRIPTION
**Pull request by betodealmeida**
### CATEGORY

Choose one

- [X] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

This PR fixes https://github.com/apache/incubator-superset/issues/8225.

### SUMMARY

When a column has `int64` integers and missing data, Pandas will cast it to `float64`, resulting in loss of precision and possibly returning incorrect numbers.

This PR fixes the bug by adding a method to the DB engine specs that returns a `dtype` based on the cursor description, currently implemented in Presto only. With the `dtype`, we can create a Pandas `Series` for each column, and create a `DataFrame` that has the proper types.

Note that in order to represent the column correctly we need to use a [nullable data type](https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html), introduced in Pandas 0.240. Unfortunately, [PyArrow is unable to serialize the resulting data frame](https://github.com/apache/arrow/issues/4168), so `msgpack` has to be disabled.

### TEST PLAN

Added unit test.

### ADDITIONAL INFORMATION

- [X] Has associated issue: https://github.com/apache/incubator-superset/issues/8225
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

@villebro @robdiciuccio

### CODE REVIEW

File: superset/dataframe.py [hunk 1]
```diff
@@ -183,7 +205,7 @@ def agg_func(cls, dtype, column_name):
         if (
             hasattr(dtype, "type")
             and issubclass(dtype.type, np.generic)
-            and np.issubdtype(dtype, np.number)
+            and dtype._is_numeric
```
Comment (betodealmeida) [inline superset/dataframe.py]: Actually, this method is Pandas specific. I'll have to use this and the previous one together.

**Comment by mistercrunch** [comment]
@robdiciuccio ^^^

**Comment by robdiciuccio** [comment]
The dataframe implementation looks good, as does the Presto engine dtype fix, but does this fully address https://github.com/apache/incubator-superset/issues/8225 if only Presto is handled? Are there other databases this fix should be implemented for (even in a separate PR)?

Also curious about your thoughts on the feasibility of the PyArrow workaround [here](https://issues.apache.org/jira/browse/ARROW-5379?focusedCommentId=16888163&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16888163).

**Comment by betodealmeida** [comment]
> The dataframe implementation looks good, as does the Presto engine dtype fix, but does this fully address #8225 if only Presto is handled? Are there other databases this fix should be implemented for (even in a separate PR)?

You're right, this only fixes Presto. I'll do a separate PR addressing the other DBs.

> Also curious about your thoughts on the feasibility of the PyArrow workaround here.

Looks like it would solve our problem, but I don't know if it would be better to monkey patch PyArrow (or if it can be done), or if we should create a light wrapper around it.

**Pull request merged**